compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/StreamingCodeDesugar.java

```diff
@@ -1651,8 +1845,13 @@ private BLangInvocation refactorInvocationWithIndexBasedArgs(BVarSymbol mapVarSy
                         (((BLangFieldBasedAccess) expression).expr.type.tag == TypeTags.STREAM)) {
                     BLangFieldBasedAccess expr = (BLangFieldBasedAccess) expression;
                     BLangExpression fieldAccessExpr = createMapVariableIndexAccessExpr(mapVarSymbol, expr);
-                    expressionList.add(createCheckedExpr(generateConversionExpr(fieldAccessExpr,
-                            expr.type, symResolver)));
+                    if (expr.type.tag == TypeTags.STRING) {
+                        // <string> is not a checked cast
+                        expressionList.add(generateConversionExpr(fieldAccessExpr, expr.type, symResolver));
+                    } else {
+                        expressionList.add(createCheckedExpr(generateConversionExpr(fieldAccessExpr,
+                                expr.type, symResolver)));
+                    }
```

Comment:
Then we have to fix in another one place as well right (where the output events are created)?

Reply:
Fixed in dc79d05